src/olympia/versions/models.py

```diff
@@ -231,10 +231,15 @@ def get_url_path(self):
             return ''
         return reverse('addons.versions', args=[self.addon.slug, self.version])
 
-    def delete(self):
+    def delete(self, hard=False):
```

Comment:
From our discussion in IRC: let's stay with `hard` as the name since it seems to apply best